ui/src/types/modules/file.type.ts

```diff
@@ -118,3 +118,11 @@ export interface LeftActionsMenu {
   disabled: boolean;
   click: () => void;
 }
+
+export interface FileSendData {
+  offSet: number;
+  size: number | undefined;
+  path: string;
+  merge?: boolean;
+  chunk?: boolean;
+}
```

Comment:
The above code was last changed to use `number` instead of `undefined`. You should make sure that the latest code uses this change. Otherwise it might break due to missing data during loading from server.

In terms of general optimization:

1. Reduce repetitive actions in the components.
   For example,
    ```css
      export interface RightPanelActionsMenu {
        click: (
          e: React.MouseEvent<HTMLButtonElement>,
          action: "send" | "download"
        ) => void;
      } 
``` 

2. Improve readability with comments and documentation on properties/props for each component.

3. Consider refactoring complex logic into helper functions, which will lead to cleaner and easier-to-read code.